src/api.coffee

```diff
@@ -185,13 +189,16 @@ module.exports = (application) ->
 		if !appId?
 			return res.status(400).send('Missing app id')
 		Promise.using application.lockUpdates(appId, true), ->
-			columns = [ 'appId', 'containerId', 'commit', 'imageId', 'env' ]
+			columns = [ 'appId', 'containerName', 'commit', 'imageId', 'env' ]
 			utils.getKnexApp(appId, columns)
 			.then (app) ->
-				# Don't return keys on the endpoint
-				app.env = _.omit(JSON.parse(app.env), config.privateAppEnvVars)
-				# Don't return data that will be of no use to the user
-				res.json(app)
+				application.getContainerId(app)
+				.then (containerId) ->
+					# Don't return keys on the endpoint
+					app.env = _.omit(JSON.parse(app.env), config.privateAppEnvVars)
+					app.containerId = containerId
+					# Don't return data that will be of no use to the user
+					res.json(_.omit(app, [ 'containerName' ]))
```

Comment:
I'd prefer a whitelist approach to rather than this blacklist approach (ie `_.pick`), also does the loss of the returned `containerId` matter? Technically it's a breaking change

Reply:
The whitelist is already implemented by selecting the columns in utils.getKnexApp above - here we just want to remove the containerName, which we purpusefully add to the selection only because we need it to get the containerId. I added the application.getContainerId function here precisely so that it's *not* a breaking change, we still send it.

Reply:
Ok